mantidproject/mantid
# Vesuvio Fitting no longer gets 'stuck' on first spectra

Vesuvio Commands script will now fit all the spectra with the correct data 

**To test:**
- **Functional Change:**  Run a Vesuvio Driver Script and ensure that all fits for the spectra are correct
  - A Vesuvio driver script can be obtained from `\\olympic\babylon5\Public\ElliotOram\Vesuvio\lid_sd_dd_gamma_ms.py`
    - If you are going to use this script it would be worth altering the `spectra` flag to something smaller (3-25) and the iterations (at the bottom) to 2. This is because the script has a long run time.
  - You can check the fit were successful by looking into the group workspaces produced and plotting:
    - data: `data_spectrum` (workspace index 0)
    - fit : `correction_spectrum_..._TotalScattering`
    - ms : `correction_spectrum_..._MultipleScattering`
    - background : `correction_spectrum_..._Background`
  - It's more important to check the later fits as they were the ones that were not fitting correctly.
- **OutputChange:** Additionally check that the output for `runs_params_iteration_x` and `runs_params_pre_correction_iteration_x` (where runs is the numbers being processed and x is the current iteration) contains a value for all parameters per spectrum. Also ensure that there is only one of each of these workspaces per iteration.

Fixes #15661

[Release Notes](https://github.com/mantidproject/mantid/blob/15661_Vesuvio_fitting_appears_to_get_stuck/docs/source/release/v3.7.0/indirect_inelastic.rst)

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
